include/inviwo/core/datastructures/volume/volumeramprecision.h

```diff
@@ -233,6 +249,19 @@ void VolumeRAMPrecision<T>::setDimensions(size3_t dimensions) {
     ownsDataPtr_ = true;
 }
 
+template <typename T>
+void VolumeRAMPrecision<T>::setSwizzleMask(const SwizzleMask& mask) {
+    swizzleMask_ = mask;
+    if (owner_) {
+        owner_->setSwizzleMask(mask);
+    }
+}
+
+template <typename T>
+SwizzleMask VolumeRAMPrecision<T>::getSwizzleMask() const {
+    return swizzleMask_;
+}
+
 template <typename T>
 double VolumeRAMPrecision<T>::getAsDouble(const size3_t& pos) const {
     return util::glm_convert<double>(data_[posToIndex(pos, dimensions_)]);
```

Comment:
Should not the swizzle-mask affect all data-getters?

Reply:
The LayerRAMPrecision does not use the swizzeling, maybe it should, but I think it would be another PR since that has a quite high impact of existing methods. Also the Volume and Image sampler should consider swizzeling

Reply:
Agree with @r-englund. That should definitely be implemented. But potentially part of different PR.